core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -80,6 +85,8 @@ public TranslogWriter(
         this.outputStream = new BufferedChannelOutputStream(java.nio.channels.Channels.newOutputStream(channel), bufferSize.bytesAsInt());
         this.lastSyncedCheckpoint = initialCheckpoint;
         this.totalOffset = initialCheckpoint.offset;
+        this.minSeqNo = initialCheckpoint.minSeqNo;
```

Comment:
Yes please. That way I don't start thinking about crazy stuff that we're not doing.